src/main/java/Duke.java

```diff
@@ -1,10 +1,121 @@
+
+        Duke.logo();
+        Duke.greeting();
+
+        Scanner input = new Scanner(System.in);
+        String line = "";
+
+        Task[] tasks = new Task[TASK_ARRAY_SIZE];
+
+        while (!(line = input.nextLine()).equals(EXIT_PROGRAM)) {
+           Duke.processInput(line, tasks);
+        }
+
+        Duke.ending();
+        System.exit(0);
+
+    }
+
+    public static void logo(){
+        System.out.println(WELCOME_MESSAGE + LOGO);
+    }
+
+    public static void greeting(){
+        System.out.println(DIVIDER + GREETINGS + DIVIDER);
+    }
+
+    public static void ending(){
+        System.out.println(ENDING + DIVIDER);
+    }
+    public static void printAddTaskMessage(Task t){
+        System.out.println(ADDED_TASK + t + "\nNow you have " + t.getNumberOfTasks() +" tasks in the list.");
+    }
+    public static void processInput(String line, Task[] tasks){
+
+        String[] words = line.split(" ",2);
+        String command = words[0];
+        int tasksCount = 0;
+        if(tasks[0]!=null){
+            tasksCount = tasks[0].getNumberOfTasks();
+        }
+        // words[0] is the command, words[n] is the next few words
+
+        switch (command) {
+        case "todo":
+            Task td = new Todo(words[1]);
+            tasks[tasksCount] = td;
+            printAddTaskMessage(td);
+            break;
+        case "deadline":
+            line = words[1]; // to remove the command
+            String[] deadlineDetails = line.split(BY_DELIMITER);
+            Task d = new Deadline(deadlineDetails[0],deadlineDetails[1]);
+            tasks[tasksCount] = d;
+            printAddTaskMessage(d);
+            break;
+        case "event":
+            line = words[1]; // to remove the command
+            String[] eventDetails = line.split(FROM_DELIMITER);
+            String eventName = eventDetails[0];
+            String from = line.substring(line.indexOf(FROM_DELIMITER) + FROM_DELIMITER.length(), line.indexOf(TO_DELIMITER));
+            eventDetails = line.split(TO_DELIMITER);
+            String to = eventDetails[1];
+            Task e = new Event(eventName,from,to);
+            tasks[tasksCount] = e;
+            printAddTaskMessage(e);
+            break;
+
+        case "mark":
+            int markIndex = Integer.parseInt(words[1]) - 1; // 0 indexing
+            System.out.println(MARKED_THIS_TASK_AS_DONE);
+            tasks[markIndex].markAsDone();
+            System.out.println(tasks[markIndex]);
+            break;
+        case "unmark":
+            int unmarkIndex = Integer.parseInt(words[1]) - 1; // 0 indexing
+            System.out.println(UNMARKED_THIS_TASK_AS_DONE);
+            tasks[unmarkIndex].markAsNotDone();
+            System.out.println(tasks[unmarkIndex]);
+            break;
+        case "list":
+            int count = 0;
+            for (Task t : tasks){
+                if(t != null){
+                    System.out.println(++count +". " + t);
```

Comment:
count could be renamed to make it clear what it does?

Reply:
Okay! will rename it to be something better :)